mlrun/config.py

```diff
@@ -846,6 +847,10 @@ def is_api_running_on_k8s(self):
         # when there is a cluster the .namespace is set
         return True if mlrun.mlconf.namespace else False
 
+    def is_nuclio_detected(self):
+        # determine is Nuclio service is detected, when the nuclio_version is not set
+        return True if mlrun.mlconf.nuclio_version else False
+
```

Comment:
I am a bit afraid of that assumption, as it is possible someone configured `nuclio_dashboard_url` but in the time the API was initializing the nuclio server wasn't yet ready or for someone reason wasn't available.

In `is_nuclio_version_in_range` we are not failing if we were unable to parse the nuclio version so we allow lazy fail when deploying the function and if the server is unavailable then we will fail.

So I think in your scenario I would prefer `nuclio_dashboard_url` instead of `nuclio_version`. WDYT